result-watcher.bal

```diff
@@ -140,6 +195,9 @@ isolated class ResultProcessor {
     }
 
     private isolated function saveErrorResult() returns error? {
+        lock {
+            _ = check removeResultWatcherFromRegistry(self.stepId);
```

Comment:
This should happen on transaction commit (https://ballerina.io/learn/by-example/commit-rollback-handlers.html?is_ref_by_example=true)
self.unschedule should also be called only on commit (even better a rollback triggers a reschedule).
At least leave a todo here for the time worker scheduling is cleaned up later.